### PR TITLE
peer/cmd: add logging-related command line options (v2)

### PR DESCRIPTION
**Pull request by nhoriguchi**
Version 2 of #44 for original issue #42. The main change is to separate option parser code from core package by applying "functional options" pattern.

### CODE REVIEW

File: sample/peer/cmd/root.go [hunk 1]
```diff
@@ -67,6 +67,14 @@ func init() {
...
 	must(viper.BindPFlag("keys",
 		rootCmd.PersistentFlags().Lookup("keys")))
+
+	rootCmd.PersistentFlags().String("logging-level", "", "logging level")
+	must(viper.BindPFlag("logging.level",
+		rootCmd.PersistentFlags().Lookup("logging-level")))
+
+	rootCmd.PersistentFlags().String("logging-file", "", "logging file")
+	must(viper.BindPFlag("logging.file",
+		rootCmd.PersistentFlags().Lookup("logging-file")))
```
Comment (sergefdrv) [inline sample/peer/cmd/root.go:77]: Adding this code to `root.go` would suggest that `peer request` command also supports those flags, whereas it doesn't. Please consider moving this code to `run.go`. Later on, we can add proper logging to the client module as well and support those flags for both `run` and `request` commands.

File: core/options.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import (
+	"os"
+
+	logging "github.com/op/go-logging"
+)
+
+type Options struct {
+	LogLevel  logging.Level
+	LogFile   *os.File
+}
```
Comment (sergefdrv) [inline core/options.go:28]: Seemingly, there is no need to make this structure exported. Consider renaming it, as well as its fields, to start with low-case letters.

**Review by sergefdrv — CHANGES_REQUESTED** [review]
Thanks a lot for your effort. I sill have some comments, but rather easy to address. This commit is relatively small, so it's okay to keep it all together. Otherwise it could also be split to (that would be too pedantic, though :laughing: ):
1. Add the mechanism to pass options when creating a replica instance
2. Support passing log options
3. Add CLI flags support to `peer` command

File: sample/peer/cmd/root.go (hunk 1 above)
Comment (nhoriguchi) [reply sample/peer/cmd/root.go:77]: Right, I'll move this to run.go.

File: core/options.go (hunk 2 above)
Comment (nhoriguchi) [reply core/options.go:28]: OK, thanks.

**Comment by nhoriguchi** [comment]
Thank you for the comments. I'll submit the updated version soon.

**Comment by nhoriguchi** [comment]
Updated with v3 (see #47).

**Pull request closed without merging**